snackbar/material/scripts/setup.sh

```diff
@@ -0,0 +1,41 @@
+#!/bin/bash
+
+# Define escape sequences for colors
+RED='\033[0;31m'
+GREEN='\033[0;32m'
+BLUE='\033[0;34m'
+CYAN='\033[0;36m'
+YELLOW='\033[0;33m'
+BOLD='\033[1m'
+RESET='\033[0m'
+
+# URL of the ZIP archive
+URL="https://inf-lab.dev/snackbar/material/lab-snackbar.zip"
+
+# the name of the zip file that has been downloaded
+ZIP_FILE="lab-snackbar.zip"
+
+# Failsafe check to not brick the local environment
+if [ -f "../../private/guide.md" ]; then
+  echo -e "${RED}⚠️  Failsafe triggered! Cannot run in local environment. Exiting script.${RESET}"
+  exit 1
+fi
```

Comment:
Diese Materialien sollen dir helfen, ein tieferes Verständnis des Kursinhalts zu erlangen und dich auf die Prüfung vorzubereiten. Wir nutzen dieses Repository ausdrücklich, um Aufgaben zu veröffentlichen, die wir gemeinsam in den Tutorien besprechen möchten.